### PR TITLE
feat: message retention policy for wakustore

### LINKED ISSUE #69: Production readiness for store protocol
- Days a message can persist
- Memory usage

### PR DESCRIPTION
**Pull request by richard-ramos**
Fixes #69

### CODE REVIEW

File: waku/node.go [hunk 1]
```diff
@@ -144,9 +144,10 @@ func Execute(options Options) {
 	}
 
 	if options.Store.Enable {
-		nodeOpts = append(nodeOpts, node.WithWakuStore(true, options.Store.ShouldResume))
+		maxDays := time.Hour * 24 * time.Duration(options.Store.RetentionMaxDays)
```
Comment (alaibe) [inline waku/node.go:147]: nit: could be a function of StoreOptionsL
```
func (s *StoreOptions) RetentionMaxDaysDuration() time.Duration
```

File: waku/persistence/store.go [hunk 2]
```diff
@@ -49,17 +54,32 @@ func WithDriver(driverName string, datasourceName string) DBOption {
...
+func WithRetentionPolicy(maxMessages int, maxDays time.Duration) DBOption {
+	return func(d *DBStore) error {
+		d.maxDays = maxDays
+		d.maxMessages = maxMessages
+		return nil
+	}
+}
+
 // Creates a new DB store using the db specified via options.
 // It will create a messages table if it does not exist
```
Comment (alaibe) [inline waku/persistence/store.go:66]: nit: and clean older records

File: waku/persistence/store.go [hunk 3]
```diff
@@ -84,6 +104,28 @@ func (d *DBStore) createTable() error {
 	return nil
 }
 
+func (d *DBStore) cleanOlderRecords() error {
```
Comment (alaibe) [inline waku/persistence/store.go:107]: shall we have a goroutine cleaning the record periodically?

File: waku/v2/protocol/store/message_queue.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package store
+
+type MessageQueue struct {
```
Comment (alaibe) [inline waku/v2/protocol/store/message_queue.go:3]: the message queue doesn t really deal with retention time? only the number of message we store.As far as I understand, the queue is here as a cache layer?

**Review by alaibe — APPROVED** [review]
Nice work!

File: waku/persistence/store.go (hunk 3 above)
Comment (richard-ramos) [reply waku/persistence/store.go:107]: Not sure, currently go-waku only fetches the records from the DB during login, and this function would have been executed before so the records returned would be valid according to the retention policy

File: waku/v2/protocol/store/message_queue.go (hunk 4 above)
Comment (richard-ramos) [reply waku/v2/protocol/store/message_queue.go:3]: Yeah, currently it only deals with number of messages. Perhaps the goroutine for cleaning older records can be added here?

**Pull request merged**